### PR TITLE
Add tower to cg workflow taxprofiler run

### PR DESCRIPTION
**Pull request by sofstam**
## Description
This PR adds tower option to `cg workflow taxprofiler run`
Open [PR](https://github.com/Clinical-Genomics/servers/pull/1098) in servers

### Added

-  Option tower in `cg workflow taxprofiler run` 

### How to prepare for test

- [x] `cg workflow taxprofiler run -d richurchin`
Screenshot 2023-08-24 at 10 48 09

- [x] `cg workflow taxprofiler run richurchin --from-start --compute-env hasta-stage`
Screenshot 2023-08-24 at 10 59 10

Screenshot 2023-08-24 at 12 13 39

Screenshot 2023-08-24 at 13 45 22

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b taxprofiler_run_tower -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by sofstam** [comment]
There is a code smell regarding the `run command` and the 13 authorized arguments. I have not solved it yet but otherwise it is ready for review :)

File: cg/models/cg_config.py [hunk 1]
```diff
@@ -177,6 +177,8 @@ class TaxprofilerConfig(CommonAppConfig):
     hostremoval_reference: str
     databases: str
     slurm: SlurmConfig
+    tower_binary_path: str
+    tower_pipeline: str
```
Comment (fevac) [inline cg/models/cg_config.py:181]: Not for this PR but we need to finish refactoring the tower and nextflow classes and the cg config file in servers so this is not duplicated.

File: cg/cli/workflow/taxprofiler/base.py [hunk 2]
```diff
@@ -125,3 +137,5 @@ def run(
     except Exception as error:
         LOG.error(f"Could not run analysis: {error}")
         raise click.Abort() from error
+    if not dry_run:
```
Comment (fevac) [inline cg/cli/workflow/taxprofiler/base.py:140]: Currently we only interact with trailblazer for tower analyses (not when using nextflow). We should make sure that every we interact with trailblazer we only do it when using tower something like:
```suggestion
    if not dry_run and with_tower:
```
Similar when accessing the tower_id for re- runs. I believe this is currently not handled in rnafusion. 
Could you test your branch for running taxprofiler with nextflow from-start and with resume, just to make sure we are handling those?

File: tests/cli/workflow/taxprofiler/test_cli_taxprofiler_run.py [hunk 3]
```diff
@@ -72,3 +72,33 @@ def test_with_config_use_nextflow(
...
+def test_with_config_use_tower(
+    cli_runner: CliRunner,
+    taxprofiler_context: CGConfig,
+    caplog: LogCaptureFixture,
+    taxprofiler_case_id: str,
+    taxprofiler_config,
+):
+    """Test command with case_id and config file using tower."""
+    caplog.set_level(logging.INFO)
+    # GIVEN case-id
```
Comment (fevac) [inline tests/cli/workflow/taxprofiler/test_cli_taxprofiler_run.py:86]:
```suggestion
    # GIVEN  a valid case
```

File: tests/cli/workflow/taxprofiler/test_cli_taxprofiler_run.py [hunk 4]
```diff
@@ -72,3 +72,33 @@ def test_with_config_use_nextflow(
...
+
+    # WHEN dry running with dry specified
+    result = cli_runner.invoke(run, [case_id, "--from-start", "--dry-run"], obj=taxprofiler_context)
+
+    # THEN command should execute successfully
+    assert result.exit_code == EXIT_SUCCESS
+
+    # THEN command should use tower
+
+    for message in [
```
Comment (fevac) [inline tests/cli/workflow/taxprofiler/test_cli_taxprofiler_run.py:99]: do we want to test all default parameters?

**Review by fevac — CHANGES_REQUESTED** [review]
Nicee!! 🌟 just minor comments from my side

**Review by diitaz93 — COMMENTED** [review]
Not an expert in the topic but looks good to me 😃

File: cg/cli/workflow/taxprofiler/base.py [hunk 5]
```diff
@@ -68,10 +75,13 @@ def config_case(
 @OPTION_WORKDIR
 @OPTION_FROM_START
 @OPTION_PROFILE
+@OPTION_TOWER
```
Comment (fevac) [inline cg/cli/workflow/taxprofiler/base.py:78]: let's remove this option. It's confusing and when we are using nextflow we do not want to use tower

File: cg/cli/workflow/taxprofiler/base.py [hunk 6]
```diff
@@ -81,10 +91,13 @@ def run(
     work_dir: str,
     from_start: bool,
     profile: str,
+    with_tower: bool,
```
Comment (fevac) [inline cg/cli/workflow/taxprofiler/base.py:94]: remove option

File: cg/cli/workflow/taxprofiler/base.py [hunk 7]
```diff
@@ -123,3 +139,5 @@ def run(
     except Exception as error:
         LOG.error(f"Could not run analysis: {error}")
         raise click.Abort() from error
+    if not dry_run and not use_nextflow:
```
Comment (fevac) [inline cg/cli/workflow/taxprofiler/base.py:142]: nice! 🌟

**Review by fevac — CHANGES_REQUESTED** [review]
nice! 🌟  good catch with the trailblazer logic. Some few additional comments

**Comment by sofstam** [comment]
@fevac I have adressed your comments now.

**Review by fevac — APPROVED** [review]
🚀 💯

**Comment by sofstam** [comment]
I will merge and deploy.

**Comment by fevac** [comment]
it seems you need an approval from sysdev. @diitaz93 do you have time for it?

**Review by diitaz93 — APPROVED** [review]
👍

**Comment by sofstam** [comment]
Deploy on stage:
Screenshot 2023-09-04 at 14 29 59

**Comment by sofstam** [comment]
Deploy on production:
Screenshot 2023-09-04 at 14 33 29

**Pull request merged**